### PR TITLE
Purge node manager

### PR DESCRIPTION
**Pull request by adauto82**

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 6 issues

1. ![CRITICAL][CRITICAL] [NodeManager.java#L31](https://github.com/rsksmart/rskj/blob/afb1a836909a74949f782a44edcefc9c238366f8/rskj-core/src/main/java/org/ethereum/net/NodeManager.java#L31): Unused import - javax.annotation.PostConstruct. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [NodeManager.java#L61](https://github.com/rsksmart/rskj/blob/afb1a836909a74949f782a44edcefc9c238366f8/rskj-core/src/main/java/org/ethereum/net/NodeManager.java#L61): Remove this unused "homeNode" private field. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1068)
1. ![CRITICAL][CRITICAL] [NodeManager.java#L65](https://github.com/rsksmart/rskj/blob/afb1a836909a74949f782a44edcefc9c238366f8/rskj-core/src/main/java/org/ethereum/net/NodeManager.java#L65): Remove this unused "inited" private field. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1068)
1. ![CRITICAL][CRITICAL] [NodeManager.java#L124](https://github.com/rsksmart/rskj/blob/afb1a836909a74949f782a44edcefc9c238366f8/rskj-core/src/main/java/org/ethereum/net/NodeManager.java#L124): Cast one of the operands of this subtraction operation to a "long". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2184)
1. ![MAJOR][MAJOR] [NodeManager.java#L65](https://github.com/rsksmart/rskj/blob/afb1a836909a74949f782a44edcefc9c238366f8/rskj-core/src/main/java/org/ethereum/net/NodeManager.java#L65): Unread field: org.ethereum.net.NodeManager.inited [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3AURF_UNREAD_FIELD)
1. ![MAJOR][MAJOR] [NodeManager.java#L73](https://github.com/rsksmart/rskj/blob/afb1a836909a74949f782a44edcefc9c238366f8/rskj-core/src/main/java/org/ethereum/net/NodeManager.java#L73): Unread field: org.ethereum.net.NodeManager.homeNode [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3AURF_UNREAD_FIELD)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by rskops** [comment]
SonarQube analysis reported 6 issues

1. ![CRITICAL][CRITICAL] [NodeManager.java#L31](https://github.com/rsksmart/rskj/blob/e7f103526b59f123096528112bb2535afd06aadd/rskj-core/src/main/java/org/ethereum/net/NodeManager.java#L31): Unused import - javax.annotation.PostConstruct. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [NodeManager.java#L61](https://github.com/rsksmart/rskj/blob/e7f103526b59f123096528112bb2535afd06aadd/rskj-core/src/main/java/org/ethereum/net/NodeManager.java#L61): Remove this unused "homeNode" private field. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1068)
1. ![CRITICAL][CRITICAL] [NodeManager.java#L65](https://github.com/rsksmart/rskj/blob/e7f103526b59f123096528112bb2535afd06aadd/rskj-core/src/main/java/org/ethereum/net/NodeManager.java#L65): Remove this unused "inited" private field. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1068)
1. ![CRITICAL][CRITICAL] [NodeManager.java#L124](https://github.com/rsksmart/rskj/blob/e7f103526b59f123096528112bb2535afd06aadd/rskj-core/src/main/java/org/ethereum/net/NodeManager.java#L124): Cast one of the operands of this subtraction operation to a "long". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2184)
1. ![MAJOR][MAJOR] [NodeManager.java#L65](https://github.com/rsksmart/rskj/blob/e7f103526b59f123096528112bb2535afd06aadd/rskj-core/src/main/java/org/ethereum/net/NodeManager.java#L65): Unread field: org.ethereum.net.NodeManager.inited [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3AURF_UNREAD_FIELD)
1. ![MAJOR][MAJOR] [NodeManager.java#L73](https://github.com/rsksmart/rskj/blob/e7f103526b59f123096528112bb2535afd06aadd/rskj-core/src/main/java/org/ethereum/net/NodeManager.java#L73): Unread field: org.ethereum.net.NodeManager.homeNode [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3AURF_UNREAD_FIELD)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by adauto82** [comment]
Rebased to 0.5.0

File: rskj-core/src/main/java/org/ethereum/net/NodeManager.java [hunk 1]
```diff
@@ -45,8 +45,8 @@
 public class NodeManager {
     private static final Logger logger = LoggerFactory.getLogger("discover");
 
-    private static final int MAX_NODES = 2000;
-    private static final int NODES_TRIM_THRESHOLD = 3000;
+    private static final long MAX_NODES = 2000;
```
Comment (lsebrie) [inline rskj-core/src/main/java/org/ethereum/net/NodeManager.java]: Here I would keep int and wouldn't use protected for NODES_TRIM_THRESHOLD

File: rskj-core/src/main/java/org/ethereum/net/NodeManager.java [hunk 2]
```diff
@@ -57,24 +57,15 @@
...
 
-    private boolean inited = false;
-
     @Autowired
     public NodeManager(PeerExplorer peerExplorer, SystemProperties config) {
         this.peerExplorer = peerExplorer;
         this.config = config;
-    }
-
-    @PostConstruct
```
Comment (lsebrie) [inline rskj-core/src/main/java/org/ethereum/net/NodeManager.java]: Great this one got erased

File: rskj-core/src/main/java/org/ethereum/net/NodeManager.java (hunk 1 above)
Comment (adauto82) [reply rskj-core/src/main/java/org/ethereum/net/NodeManager.java]: Sonar asked for one of terms of the subtraction to be a long.
I need to access the value of NODES_TRIM_THRESHOLD from test. would you be satisfied with no keyword?

File: rskj-core/src/main/java/org/ethereum/net/NodeManager.java [hunk 3]
```diff
@@ -119,24 +108,17 @@ public NodeStatistics getNodeStatistics(Node n) {
...
-                if (nodeHandlerMap.size() <= MAX_NODES) {
-                    break;
-                }
-            }
+            //I create a stream
+            List<NodeHandler> toRemove = nodeHandlerMap.values().stream()
+                    //sort by reputation
+                    .sorted(Comparator.comparingInt(o -> o.getNodeStatistics().getReputation()))
+                    //and just keep the ones that exceeds the MAX_NODES
+                    .limit(nodeHandlerMap.size() - MAX_NODES)
```
Comment (juli) [inline rskj-core/src/main/java/org/ethereum/net/NodeManager.java]: If you have 3001 nodes you want to keep 1001 or 2000?

Comment (amendelzon) [reply rskj-core/src/main/java/org/ethereum/net/NodeManager.java]: :+1:

**Pull request merged**